src/client/src/producer.rs

```diff
@@ -17,16 +17,13 @@ pub struct Producer {
 }
 
 impl Producer {
-    pub fn new(replica: ReplicaKey, pool: SpuPool) -> Self {
+    pub(crate) fn new(replica: ReplicaKey, pool: SpuPool) -> Self {
         Self { replica, pool }
     }
 
-    pub fn replica(&self) -> &ReplicaKey {
-        &self.replica
-    }
-
     /// send records to spu leader for replica
-    pub async fn send_record(&mut self, record: Vec<u8>) -> Result<(), ClientError> {
+    pub async fn send_record<B: AsRef<[u8]>>(&mut self, buffer: B) -> Result<(), FluvioError> {
+        let record = buffer.as_ref().to_owned();
```

Comment:
this creates intermediate allocation.  pass references down rather than trying to own it.